### PR TITLE
[FLINK-36689][Runtime/Web Frontend] Update ng-zorro-antd to v18

### PR DESCRIPTION
**Pull request by simplejason**
## What is the purpose of the change
Related-to: https://issues.apache.org/jira/browse/FLINK-36689
Angular framework and NG-ZORRO both have released their stable v18 versions. It is a good time to bump them to the latest.

## Brief change log

- Angular: v18.2.13
- ng-zorro-antd: v18.2.1
- typescript: v5.4.5

## Verifying this change

Please make sure both new and modified tests in this PR follow [the conventions for tests defined in our code quality guide](https://flink.apache.org/how-to-contribute/code-style-and-quality-common/#7-testing).

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluster with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

**Comment by simplejason** [comment]
Error reported. It may be due to linux version issue, it's being solved.
```bash
12:25:18.540 [INFO] --- frontend-maven-plugin:1.12.1:install-node-and-npm (install node and npm) @ flink-runtime-web ---
12:25:20.411 [INFO] Installing node version v18.19.1
12:25:20.411 [INFO] Downloading https://nodejs.org/dist/v18.19.1/node-v18.19.1-linux-x64.tar.gz to /__w/2/.m2/repository/com/github/eirslett/node/18.19.1/node-18.19.1-linux-x64.tar.gz
12:25:20.413 [INFO] No proxies configured
12:25:20.413 [INFO] No proxy was configured, downloading directly
12:25:23.160 [INFO] Unpacking /__w/2/.m2/repository/com/github/eirslett/node/18.19.1/node-18.19.1-linux-x64.tar.gz into /__w/2/s/flink-runtime-web/web-dashboard/node/tmp
12:25:25.328 [INFO] Copying node binary from /__w/2/s/flink-runtime-web/web-dashboard/node/tmp/node-v18.19.1-linux-x64/bin/node to /__w/2/s/flink-runtime-web/web-dashboard/node/node
12:25:25.758 [INFO] Installed node locally.
12:25:25.760 [INFO] Installing npm version 10.2.4
12:25:25.760 [INFO] Downloading https://registry.npmjs.org/npm/-/npm-10.2.4.tgz to /__w/2/.m2/repository/com/github/eirslett/npm/10.2.4/npm-10.2.4.tar.gz
12:25:25.760 [INFO] No proxies configured
12:25:25.760 [INFO] No proxy was configured, downloading directly
12:25:25.818 [INFO] Unpacking /__w/2/.m2/repository/com/github/eirslett/npm/10.2.4/npm-10.2.4.tar.gz into /__w/2/s/flink-runtime-web/web-dashboard/node/node_modules
12:25:26.241 [INFO] Installed npm locally.
12:25:26.242 [INFO] 
12:25:26.242 [INFO] --- frontend-maven-plugin:1.12.1:npm (npm install) @ flink-runtime-web ---
12:25:26.246 [INFO] Running 'npm ci --cache-max=0 --no-save ${npm.proxy}' in /__w/2/s/flink-runtime-web/web-dashboard
12:25:26.261 [INFO] /__w/2/s/flink-runtime-web/web-dashboard/node/node: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by /__w/2/s/flink-runtime-web/web-dashboard/node/node)
12:25:26.261 [INFO] /__w/2/s/flink-runtime-web/web-dashboard/node/node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by /__w/2/s/flink-runtime-web/web-dashboard/node/node)
12:25:26.261 [INFO] /__w/2/s/flink-runtime-web/web-dashboard/node/node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /__w/2/s/flink-runtime-web/web-dashboard/node/node)
12:25:26.263 [INFO] 

```

File: flink-runtime-web/pom.xml [hunk 1]
```diff
@@ -263,8 +263,8 @@ under the License.
 							<goal>install-node-and-npm</goal>
 						</goals>
 						<configuration>
-							<nodeVersion>v16.13.2</nodeVersion>
-							<npmVersion>8.1.2</npmVersion>
+							<nodeVersion>v18.19.1</nodeVersion>
```
Comment (Laffery) [inline flink-runtime-web/pom.xml:266]: Maintenance of node.js v18 LTS will end in May 2025, what about using v20 LTS directly?

Comment (Laffery) [reply flink-runtime-web/pom.xml:266]: > Maintenance of node.js v18 LTS will end in May 2025, what about using v20 LTS directly?

ref https://nodejs.org/en/about/previous-releases

**Comment by simplejason** [comment]
@flinkbot run azure

**Review by Laffery — COMMENTED** [review]
In tsconfig.json

```diff
"compilerOptions": {
- "module": "es2020",
- "target": "es2020",
+ "module": "es2022",
+ "target": "es2022",
+ "useDefineForClassFields": false,
}

**Comment by simplejason** [comment]
@flinkbot run azure

**Comment by simplejason** [comment]
@flinkbot run azure

**Comment by simplejason** [comment]
All the code for the upgrade has been committed, and this pr does not contain node-related changes, as the node version has already been upgraded in #25670. @reswqa Can you re-run the last Azure build? It seems that `@flinkbot run azure` not work :( cc @MartijnVisser

**Review by Laffery — APPROVED** [review]
LGTM

**Comment by davidradl** [comment]
> LGTM

the CI tests still show FAILURE

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Pull request merged**